### PR TITLE
Added apiCallTimeout param for Bedrock and it fixes langchain4j/langc…

### LINKED ISSUE #1704: [FEATURE] Add timeout support for Amazon Bedrock Anthropic API
**Is your feature request related to a problem? Please describe.**
The **BedrockAnthropicMessageChatModel** in LangChain4j currently lacks a **timeout()** method, which is available in other model implementations like **AzureOpenAiChatModel**. This inconsistency makes it difficult to set timeouts for API calls when using the Amazon Bedrock Anthropic API, potentially leading to long-running requests that could impact application performance.

**Describe the solution you'd like**
I propose adding a **timeout()** method to the **BedrockAnthropicMessageChatModel** builder, similar to the one available in **AzureOpenAiChatModel**. This would allow users to set a timeout duration for API calls, enhancing control over request handling. The implementation should be consistent with other LangChain4j model implementations. For example:

```java
BedrockAnthropicMessageChatModel.builder()
    .region(Region.US_EAST_1)
    .model("anthropic.claude-3-5-sonnet-20240620-v1:0")
    .temperature(0.3)
    .topP(0.3f)
    .timeout(Duration.ofSeconds(60))  // New timeout method
    .build()
```

**Describe alternatives you've considered**
1. Using the AWS SDK v2 directly, which provides timeout functionality through the **ApacheHttpClient** configuration. However, this approach requires more boilerplate code and deviates from the simplified LangChain4j API.

2. Implementing a wrapper class around **BedrockAnthropicMessageChatModel** to add timeout functionality. This solution, while feasible, would not be as clean or consistent with other LangChain4j implementations.

**Additional context**
The AWS CLI and SDK v2 both support setting timeouts for Bedrock API calls. Implementing this feature in LangChain4j would bring the library's functionality in line with these official AWS tools, providing a more consistent developer experience across the ecosystem.

For reference, here's how timeouts are set in the AWS CLI:

```bash
aws bedrock-runtime invoke-model \
    --model-id anthropic.claude-3-5-sonnet-20240620-v1:0 \
    --region us-east-1 \
    --body "$(echo '{
        "anthropic_version": "bedrock-2023-05-31",
        "max_tokens": 200000,
        "messages": [
            {
                "role": "user",
                "content": "Hello!"
            }
        ],
        "temperature": 0.3,
        "top_p": 0.3
    }' | base64 -w 0)" \
    --content-type "application/json" \
    --cli-read-timeout 60 \
    --cli-connect-timeout 60 \
    output.json
```

And in the AWS SDK v2:

```java
Region region = Region.US_EAST_1;
String modelId = "anthropic.claude-3-5-sonnet-20240620-v1:0";
ApacheHttpClient httpClient = ApacheHttpClient.builder()
        .connectionTimeout(Duration.ofSeconds(60))
        .socketTimeout(Duration.ofSeconds(60))
        .build();

BedrockRuntimeClient bedrockRuntimeClient = BedrockRuntimeClient.builder()
        .region(region)
        .credentialsProvider(DefaultCredentialsProvider.create())
        .httpClient(httpClient)
        .build();

Map<String, Object> requestBody = new HashMap<>();
requestBody.put("anthropic_version", "bedrock-2023-05-31");
requestBody.put("max_tokens", 200000);

List<Map<String, String>> messages = new ArrayList<>();
Map<String, String> message = new HashMap<>();
message.put("role", "user");
message.put("content", "Hello!");
messages.add(message);
requestBody.put("messages", messages);
requestBody.put("temperature", 0.3);
requestBody.put("top_p", 0.3);

ObjectMapper objectMapper = new ObjectMapper();
String jsonBody = "";
try {
    jsonBody = objectMapper.writeValueAsString(requestBody);
} catch (Exception e) {
    System.out.println("Error serializing request body: " + e.getMessage());
    return;
}

InvokeModelRequest request = InvokeModelRequest.builder()
        .modelId(modelId)
        .contentType("application/json")
        .accept("application/json")
        .body(SdkBytes.fromUtf8String(jsonBody))
        .build();

String responseBody = null;
try {
    InvokeModelResponse response = bedrockRuntimeClient.invokeModel(request);
    responseBody = response.body().asUtf8String();
} catch (Exception e) {
    System.out.println("Error invoking Bedrock model: " + e.getMessage());
} finally {
    bedrockRuntimeClient.close();
}
```

Adding this feature would significantly improve the usability and flexibility of the **BedrockAnthropicMessageChatModel** in LangChain4j. :)

### PR DESCRIPTION
**Pull request by n3kh3l**
## Issue
Fixes #1704 

## Change
Added timeout support in BedrockCharModel.  The default timeout is set to 1 minute.

## General checklist

- [x] There are no breaking changes
- [x] I have added unit and integration tests for my change
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [?] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green
I only ran core module, and I didn't run main as I don't have openAiApiKey 

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new model integration

- [ ] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractSharedBedrockChatModel.java [hunk 1]
```diff
@@ -47,6 +49,8 @@ public abstract class AbstractSharedBedrockChatModel {
     @Builder.Default
     protected final int topK = 250;
     @Builder.Default
+    protected final Duration timeout = Duration.ofMinutes(5L);
```
Comment (langchain4j) [inline langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractSharedBedrockChatModel.java]: 5 minutes seems too high... What about 1 minute?

**Review by ghost — COMMENTED** [review]
@niksbansode thank you!

File: langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractSharedBedrockChatModel.java (hunk 1 above)
Comment (forge-lit) [reply langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractSharedBedrockChatModel.java]: Done

Comment (n3kh3l) [reply langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractSharedBedrockChatModel.java]: Done

**Review by ghost — APPROVED** [review]
@niksbansode thank you!

**Pull request merged**